File: backend/backend.Domain/src/Entities/Payment.cs
using System.Text.Json.Serialization;

namespace backend.Domain.src.Entities
{
    public class Payment : BaseEntity
    {
        public PaymentMethod PaymentMethod { get; set; }
        public float TotalPrice { get; set; }
        public Guid OrderId { get; set; }

        public Order Order { get; set; }
    }
    
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum PaymentMethod
    {
        Card,
        PayPal,
        OnlinePayment
    }
}

File: frontend/src/types/Payment.ts
import { Guid } from "guid-typescript";

export interface PaymentRead 
{
    id: Guid
    paymentMethod: "Card" | "PayPal" | "OnlinePayment"
    totalPrice: number
    orderId: Guid
}

export interface PaymentCreate
{
    paymentMethod: "Card" | "PayPal" | "OnlinePayment"
    totalPrice: number
    orderId: Guid
}